File: ios/Playground/Sources/Theme/Typography.swift
import SwiftUI

// MARK: - Playground typography

/// Shantell Sans (SIL OFL) — the hand-drawn marker family behind every type role,
/// mirroring the Android design system's scale. The font is a variable TTF anchored
/// on its default instance; `.weight()` drives the `wght` axis on iOS.
/// Font + licence: `Sources/Resources/Fonts/ShantellSans.ttf`, `licenses/ShantellSans-OFL.txt`.
enum PlaygroundFont {
    /// PostScript name of the vendored variable font's default instance.
    static let baseName = "ShantellSans-Light"

    /// `relativeTo` anchors each role to its semantic `TextStyle` so the font
    /// follows that role's Dynamic Type / accessibility scaling curve, not the
    /// default body curve.
    static func shantell(_ size: CGFloat, _ weight: Font.Weight, relativeTo textStyle: Font.TextStyle) -> Font {
        Font.custom(baseName, size: size, relativeTo: textStyle).weight(weight)
    }
}

/// Type roles mirroring the Android `AutoMobileTypography` scale (Material sizes,
/// Shantell Sans family), each anchored to a matching SwiftUI `TextStyle` for
/// correct Dynamic Type scaling.
struct PlaygroundTypography {
    let displayLarge = PlaygroundFont.shantell(57, .regular, relativeTo: .largeTitle)
    let displayMedium = PlaygroundFont.shantell(45, .regular, relativeTo: .largeTitle)
    let displaySmall = PlaygroundFont.shantell(36, .regular, relativeTo: .largeTitle)
    let headlineLarge = PlaygroundFont.shantell(32, .semibold, relativeTo: .title)
    let headlineMedium = PlaygroundFont.shantell(28, .semibold, relativeTo: .title2)
    let headlineSmall = PlaygroundFont.shantell(24, .semibold, relativeTo: .title3)
    let titleLarge = PlaygroundFont.shantell(22, .medium, relativeTo: .title3)
    let titleMedium = PlaygroundFont.shantell(16, .medium, relativeTo: .headline)
    let titleSmall = PlaygroundFont.shantell(14, .medium, relativeTo: .subheadline)
    let bodyLarge = PlaygroundFont.shantell(16, .regular, relativeTo: .body)
    let bodyMedium = PlaygroundFont.shantell(14, .regular, relativeTo: .callout)
    let bodySmall = PlaygroundFont.shantell(12, .regular, relativeTo: .footnote)
    let labelLarge = PlaygroundFont.shantell(14, .medium, relativeTo: .subheadline)
    let labelMedium = PlaygroundFont.shantell(12, .medium, relativeTo: .caption)
    let labelSmall = PlaygroundFont.shantell(11, .medium, relativeTo: .caption2)
}

File: android/playground/design/system/src/main/kotlin/dev/jasonpearson/automobile/design/system/theme/Typography.kt
package dev.jasonpearson.automobile.design.system.theme

import androidx.compose.material3.Typography
import androidx.compose.ui.text.ExperimentalTextApi
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.Font
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontVariation
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.sp
import dev.jasonpearson.automobile.design.assets.R

// Shantell Sans (SIL OFL) — the hand-drawn marker family behind every type role.
// The variable font's weight axis is driven per-weight; on API < 26 the variation
// settings are ignored and the font renders at its default instance (still
// Shantell Sans, just a single weight). Font file + licence:
// design/assets/src/main/res/font/{shantell_sans.ttf, OFL.txt}.
@OptIn(ExperimentalTextApi::class)
private fun shantell(weight: FontWeight, axisWeight: Int) =
  Font(
    resId = R.font.shantell_sans,
    weight = weight,
    variationSettings = FontVariation.Settings(FontVariation.weight(axisWeight)),
  )

val ShantellSans =
  FontFamily(
    shantell(FontWeight.Normal, 400),
    shantell(FontWeight.Medium, 500),
    shantell(FontWeight.SemiBold, 600),
    shantell(FontWeight.Bold, 700),
  )

// AutoMobile Design System Typography
val AutoMobileTypography =
  Typography(
    displayLarge =
      TextStyle(
        fontFamily = ShantellSans,
        fontWeight = FontWeight.Normal,
        fontSize = 57.sp,
        lineHeight = 64.sp,
        letterSpacing = (-0.25).sp,
      ),
    displayMedium =
      TextStyle(
        fontFamily = ShantellSans,
        fontWeight = FontWeight.Normal,
        fontSize = 45.sp,
        lineHeight = 52.sp,
        letterSpacing = 0.sp,
      ),
    displaySmall =
      TextStyle(
        fontFamily = ShantellSans,
        fontWeight = FontWeight.Normal,
        fontSize = 36.sp,
        lineHeight = 44.sp,
        letterSpacing = 0.sp,
      ),
    headlineLarge =
      TextStyle(
        fontFamily = ShantellSans,
        fontWeight = FontWeight.Normal,
        fontSize = 32.sp,
        lineHeight = 40.sp,
        letterSpacing = 0.sp,
      ),
    headlineMedium =
      TextStyle(
        fontFamily = ShantellSans,
        fontWeight = FontWeight.Normal,
        fontSize = 28.sp,
        lineHeight = 36.sp,
        letterSpacing = 0.sp,
      ),
    headlineSmall =
      TextStyle(
        fontFamily = ShantellSans,
        fontWeight = FontWeight.Normal,
        fontSize = 24.sp,
        lineHeight = 32.sp,
        letterSpacing = 0.sp,
      ),
    titleLarge =
      TextStyle(
        fontFamily = ShantellSans,
        fontWeight = FontWeight.Normal,
        fontSize = 22.sp,
        lineHeight = 28.sp,
        letterSpacing = 0.sp,
      ),
    titleMedium =
      TextStyle(
        fontFamily = ShantellSans,
        fontWeight = FontWeight.Medium,
        fontSize = 16.sp,
        lineHeight = 24.sp,
        letterSpacing = 0.15.sp,
      ),
    titleSmall =
      TextStyle(
        fontFamily = ShantellSans,
        fontWeight = FontWeight.Medium,
        fontSize = 14.sp,
        lineHeight = 20.sp,
        letterSpacing = 0.1.sp,
      ),
    bodyLarge =
      TextStyle(
        fontFamily = ShantellSans,
        fontWeight = FontWeight.Normal,
        fontSize = 16.sp,
        lineHeight = 24.sp,
        letterSpacing = 0.5.sp,
      ),
    bodyMedium =
      TextStyle(
        fontFamily = ShantellSans,
        fontWeight = FontWeight.Normal,
        fontSize = 14.sp,
        lineHeight = 20.sp,
        letterSpacing = 0.25.sp,
      ),
    bodySmall =
      TextStyle(
        fontFamily = ShantellSans,
        fontWeight = FontWeight.Normal,
        fontSize = 12.sp,
        lineHeight = 16.sp,
        letterSpacing = 0.4.sp,
      ),
    labelLarge =
      TextStyle(
        fontFamily = ShantellSans,
        fontWeight = FontWeight.Medium,
        fontSize = 14.sp,
        lineHeight = 20.sp,
        letterSpacing = 0.1.sp,
      ),
    labelMedium =
      TextStyle(
        fontFamily = ShantellSans,
        fontWeight = FontWeight.Medium,
        fontSize = 12.sp,
        lineHeight = 16.sp,
        letterSpacing = 0.5.sp,
      ),
    labelSmall =
      TextStyle(
        fontFamily = ShantellSans,
        fontWeight = FontWeight.Medium,
        fontSize = 11.sp,
        lineHeight = 16.sp,
        letterSpacing = 0.5.sp,
      ),
  )
